KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,367 @@
-Current update status: incomplete
-Goals:change the book chapter files to include paragraphs from the addendum, and after that insert some edits so they are intelligently integrated into the chapter
-
-Edward Bian
-Currently under heavy modification; sections may not work and/or look inefficient/confusing
-"""
+# Path to data directory
+DATA_DIR = os.path.dirname(os.path.realpath(__file__)) + "/../data/"
+
+
 
 # start out by reading KLSadd.tex to get all of the paragraphs that must be added to the chapter files
 # also keep track of which chapter each one is in
 
-# variables: chapNums for the chapter number each section belongs to paras will hold the sections that will be copied over
+
 # chapNums is needed to know which file to open (9 or 14)
-# mathPeople is just the name for the sections that are used, like Wilson,
-# Racah, etc. I know some are not people, its just a var name
-chapNums = []
+# mathPeople is just the name for the sections that are used, like Wilson, Racah, etc.
+chap_nums = []
 paras = []
-klsparas = []
-mathPeople = []
-newCommands = []  # used to hold the indexes of the commands
-ref9II = []  # Hold section search indexes
-ref14II = []
-ref9III = []  # Holds all indexes
-ref14III = []
+
+# klsaddparas contains the locations of things with "paragraph{" in KLSadd, these are subsections that need to be sorted
+klsaddparas = []
+math_people = []
+new_commands = []  # used to hold the indexes of the commands
+
+# ref9II holds the beginnings and ends of each chapter 9 section, provided and area to search
+ref9_2 = []
+
+# ref14II holds the beginnings and ends of each chapter 14 section, provided and area to search
+ref14_2 = []
+
+# ref9III holds all subsections in each chapter 9 section along with the what  ref9II holds
+ref9_3 = []
+
+# ref14III holds all subsections in each chapter 14 section along with the what ref14II holds
+ref14_3 = []
+
+# specref9 is the KLS indexes that go in chapter 9
 specref9 = []
+
+# specref14 is the KLS indexes that in chapter 14
 specref14 = []
-comms = ""  # holds the ACTUAL STRINGS of the commands
+
+check = []
+check2 = []
+check3 = []
+kls_list = []
+sortmatch = []
+
+w, h = 3, 9999
+sorterIIIcheck = [[0 for x in range(w)] for y in range(h)]
+
+sortmatch_2 = []
+
+test = "memesbananashgvlksrjc%/n"
+print test[-2]
 # 2/18/16 this method addresses the goal of hardcoding in the necessary packages to let the chapter files run as pdf's.
-# Currently only works with chapter 9, ask Dr. Cohl to help port your
-# chapter 14 output file into a pdf
+# Currently only works with chapter 9, ask Dr. Cohl to help port your chapter 14 output file into a pdf
+
+# 5/5/16 this method implements the Hypergeometric paragraphs into relevant sections for practicality
+# Later versions should attempt to: fix paragraphs with variations on this name like
+# Basic Hypergeometric Representation, q-Hypergeometric Representation
+
+
+def new_keywords(kls):
```

Comment:
This whole function loops through a list too many times to do what it does - one iteration through the list should be able to do everything at once, since there is no modification of elements that is dependent on the new state after any pass through